prereise/gather/demanddata/transportation_electrification/immediate.py

```diff
@@ -208,33 +208,107 @@ def immediate_charging(
     flag_translation = {1: "weekend", 2: "weekday"}
     for i, weekday_flag in enumerate(input_day):
         daily_profile = daily_resampled_profiles[flag_translation[weekday_flag]]
-        if i == len(input_day) - 1:
-            # for the last day of the year, we need to wrap our 48 hours back around
-            # first half: last day of the year
-            model_year_profile[-24:] += daily_profile[:24]
-            # last half: first day of the year
-            model_year_profile[:24] += daily_profile[24:]
-        else:
-            model_year_profile[i * 24 : i * 24 + 48] += daily_profile
+
+        # create wrap-around indexing function
+        trip_window_indices = np.arange(i * 24, i * 24 + 72) % len(model_year_profile)
+
+        # MW
+        model_year_profile[trip_window_indices] += daily_profile
+
     # Normalize the output so that it sums to 1
     summed_profile = model_year_profile / model_year_profile.sum()
 
     return summed_profile
 
 
-def adjust_bev(hourly_profile, adjustment_values):  # noqa: N802
+def adjust_bev(
+    hourly_profile,
+    adjustment_values,
+    model_year,
+    veh_type,
+    veh_range,
+    bev_vmt,
+    charging_efficiency,
+):  # noqa: N802
```

Comment:
Not sure we need `noqa: N802` here given the function name is all in lower case.